### DISCUSSION TITLE
Secret do terraforma w pipeline azure devops w tasku AzureCLI@2 - z keyvaulta? z secret variable pipelinu?

### DISCUSSION THREAD

**Discussion by maciej-skorupka** [AzureCommunityPL/group#393] [Pomoc] [answered]
Hej,

Jest sobie pipeline azure devops, który ma odpalać terraforma i w jakimś tam przypadku musi podać secret jako -var do terraforma. Do tej pory używałem Variables pipelinu do ustawiania takich secretów i odczytywałem je w taskach powershell i bash 
 przez env: (https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#secret-variables). 

W aktualnym przypadku, który mam muszę użyć AzureCLI@2. No i jego parametr arguments niestety nie potrafi wciągnąć takich secretów - zawsze dostaję null - wygląda na feature request na to https://github.com/microsoft/azure-pipelines-tasks/issues/15415. I zastanawiam się nad Waszą opinią o tym, żeby w takich sytuacjach w tasku AzureCLI@2 po prostu w kodzie odwoływać się do keyvaulta (az keyvault secret show) w którym bym miał ten secret. Brzmi bezpiecznie? Można sensowniej w AzureCLI@2?

**Comment by piotrgwiazda** [comment, marked as answer]
Piszesz o Terraform - w samym tf możesz pobrać secret z KeyVault za pomocą
data i coś z nim dalej zrobić.  Tożsamość Terraforma musi mieć wtedy dostęp.

Z az cli też możesz oskryptować. Możesz się wtedy pokusić nawet o
skorzystanie z managed identity przypisanego do agenta i w ogóle nie mieć
żadnych secretów w Azure DevOps.

wt., 1 lut 2022, 17:45 użytkownik maciej-skorupka ***@***.***>
napisał:

**Reply by maciej-skorupka** [reply]
Genialne. Nigdy o tym nie pomyślałem, choć teraz wydaje się to tak oczywiste. Dzięki.
